analyses/optimus/process_10x_v2_datasets.ipynb

```diff
@@ -0,0 +1,652 @@
+    "\n",
+    "    specific_inputs = {\n",
+    "        \"Optimus.sample_id\": s.replace('/', '_'),\n",
+    "        \"Optimus.r1\": r1,\n",
+    "        \"Optimus.r2\": r2\n",
+    "    }\n",
+    "    \n",
+    "    # build the sample inputs\n",
+    "    inputs_dict = {**static_inputs, **g, **specific_inputs}\n",
+    "    with open(testdir + '/inputs/{}_inputs.json'.format(s.replace('/', '_')), 'w') as f:\n",
+    "        json.dump(inputs_dict, f)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "# this is the main pipeline wdl\n",
+    "wdl = '../../../skylab/pipelines/optimus/Optimus.wdl'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "## submit everything to cromwell"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import os\n",
+    "from google.cloud import storage\n",
+    "import cromwell_manager as cwm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "storage_client = storage.Client(project='broad-dsde-mint-dev')\n",
+    "with open(os.path.expanduser('~/.ssh/mint_cromwell_config.json')) as f:\n",
+    "    cromwell_server = cwm.Cromwell(**json.load(f))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "input_dir = testdir + '/inputs/'\n",
+    "inputs = os.listdir(input_dir)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "workflows = {}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 149,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "for input_json in inputs:\n",
+    "    name = input_json.rpartition('_')[0].split('/')[-1]\n",
+    "    workflows[name] = cwm.Workflow.from_submission(\n",
+    "        wdl=wdl, \n",
+    "        inputs_json=input_dir + input_json, \n",
+    "        cromwell_server=cromwell_server,\n",
+    "        workflow_dependencies=dependencies_zip,\n",
+    "        storage_client=storage_client)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 156,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "inputs_part2 = inputs[:5] + inputs[6:9] + inputs[11:14]"
```

Comment:
Why slicing in this way? `:
inputs_part2 = inputs[:5] + inputs[6:9] + inputs[11:14]`

Reply:
Residual metadata (some were failing and I needed to rerun). Will remove. Thanks for catching.